mopidy_local_images/library.py

```diff
@@ -179,7 +179,7 @@ def _extract_images(self, uri, tags):
                 logger.warn('Error extracting images for %r: %r', uri, e)
         # look for external album art
         path = translator.local_uri_to_path(uri, self.media_dir)
-        dirname = os.path.dirname(path)
+        dirname = os.path.dirname(path).decode('utf-8').replace('[', '[[]')
```

Comment:
Why the `decode('utf-8')`?

Reply:
And why would this be necessary? And what would happen if your local filesystem encoding was _not_ UTF-8?

Reply:
You can try yourself with some path with characters other than ASCII, I didn't find other solution than that to handle that type of characters.
Maybe it is better to use `sys.getfilesystemencoding()` instead of a fixed UTF-8 (and maybe encoding back after the replacement).

Reply:
I have to admit, I haven't tried it (haven't used any of the `mopidy-local-*` stuff in a long time...), but what's wrong with passing plain `bytes` to `replace()`, e.g.
```
dirname = os.path.dirname(path).replace(b'[', b'[[]')
```
Shouldn't that do the trick, without any need to encode/decode anything?

Reply:
I think so, when i tried it I missed the `b''` prefix.

Reply:
Mopidy runs on Python 2.7, so anything regarding python >= 3.4 is irrelevant for now.

Reply:
And Mopidy uses byte strings (hopefully) in the local filesystem encoding for `path`, so `os.path.dirname` will also return `bytes`. The main issue with the origin PR was that it implies a utf-8 filesystem encoding and would fail miserably if another encoding was used.

Reply:
Ok, got it. Well, it doesn't state so explicitly in the [docs](https://docs.mopidy.com/en/latest/modules/local/#mopidy.local.translator.local_uri_to_path), but `mopidy.local.translator.local_uri_to_path` always returns a `bytes` object, and I'm pretty sure will continue to do so unless Mopidy is ported to Python 3.